drivers/dma/xdmac_sam.c

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright (c) 2017 comsuisse AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief Atmel SAM MCU family Direct Memory Access (XDMAC) driver.
+ */
+
+#include <errno.h>
+#include <misc/__assert.h>
+#include <device.h>
+#include <init.h>
+#include <string.h>
+#include <soc.h>
+#include <dma.h>
+#include "xdmac_sam.h"
+
+#define SYS_LOG_DOMAIN "dev/xdmac_sam"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_DMA_LEVEL
+#include <logging/sys_log.h>
+
+#define XDMAC_INT_ERR (XDMAC_CIE_RBIE | XDMAC_CIE_WBIE | XDMAC_CIE_ROIE)
+#define DMA_CHANNELS_NO  XDMACCHID_NUMBER
+
+/* DMA channel configuration */
+struct xdmac_sam_channel_cfg {
+	dma_callback callback;
+};
+
+/* Device constant configuration parameters */
+struct xdmac_sam_dev_cfg {
+	Xdmac *regs;
+	void (*irq_config)(void);
+	u8_t periph_id;
+	u8_t irq_id;
+};
+
+/* Device run time data */
+struct xdmac_sam_dev_data {
+	struct xdmac_sam_channel_cfg dma_channels[DMA_CHANNELS_NO];
+};
+
+#define DEV_NAME(dev) ((dev)->config->name)
+#define DEV_CFG(dev) \
+	((const struct xdmac_sam_dev_cfg *const)(dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct xdmac_sam_dev_data *const)(dev)->driver_data)
+
+static void xdmac_sam_isr(void *arg)
+{
+	struct device *dev = (struct device *)arg;
+	const struct xdmac_sam_dev_cfg *const dev_cfg = DEV_CFG(dev);
+	struct xdmac_sam_dev_data *const dev_data = DEV_DATA(dev);
+	Xdmac *const xdmac = dev_cfg->regs;
+	struct xdmac_sam_channel_cfg *channel_cfg;
+	u32_t isr_status;
+	u32_t err;
+
+	/* Get global interrupt status */
+	isr_status = xdmac->XDMAC_GIS;
+
+	for (int channel = 0; channel < DMA_CHANNELS_NO; channel++) {
+		if (!(isr_status & (1 << channel))) {
+			continue;
+		}
+
+		channel_cfg = &dev_data->dma_channels[channel];
+
+		/* Get channel errors */
+		err = xdmac->XDMAC_CHID[channel].XDMAC_CIS & XDMAC_INT_ERR;
+
+		/* Execute callback */
+		if (channel_cfg->callback) {
+			channel_cfg->callback(dev, channel, err);
+		}
+	}
+}
+
+int xdmac_sam_channel_configure(struct device *dev, u32_t channel,
+				struct xdmac_sam_channel_config *param)
+{
+	const struct xdmac_sam_dev_cfg *const dev_cfg = DEV_CFG(dev);
+	Xdmac *const xdmac = dev_cfg->regs;
+
+	if (channel >= DMA_CHANNELS_NO)
+		return -EINVAL;
```

Comment:
add {} even on one-liner statement for if/while/for. Apply that everywhere.

That said, it's weird that checkpatch did not catch those and make the checks failing...

Reply:
Sorry, I overlooked this one.